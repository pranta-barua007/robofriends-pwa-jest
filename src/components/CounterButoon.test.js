import { shallow } from "enzyme";
import React from "react";
import CounterButton from "./CounterButton";

describe('CounterButton', () => {
    it('should render CounterButton & match snapshot', () => {
        const mockColor = 'red';
        expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
    })

    it('should recive red as value of color prop', () => {
        const mockColor = 'red';
        const wrapper = shallow(<CounterButton color={mockColor} />);

        expect(wrapper.props().color).toEqual('red');
    })

    it('should increase counter by click event', () => {
        const mockColor = 'red';
        const wrapper = shallow(<CounterButton color={mockColor} />);
        
        wrapper.find("[id='counter']").simulate('click');
        wrapper.find("[id='counter']").simulate('click');
        expect(wrapper.state()).toEqual({ count: 2 });
        wrapper.find("[id='counter']").simulate('keypress');
        expect(wrapper.state()).toEqual({ count: 2 });
    })
});