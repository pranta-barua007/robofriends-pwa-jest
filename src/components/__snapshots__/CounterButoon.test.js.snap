// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton should render CounterButton & match snapshot 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
